index.js

```diff
@@ -74,9 +75,16 @@ export default class LinterJSCS {
     require('atom-package-deps').install();
 
     this.observer = atom.workspace.observeTextEditors((editor) => {
-      editor.getBuffer().onDidSave(() => {
-        if (grammarScopes.indexOf(editor.getGrammar().scopeName) !== -1 && this.fixOnSave) {
-          this.fixString(editor);
+      editor.getBuffer().onWillSave(() => {
+
+        // Exclude `excludeFiles` for fix on save
+        const config = this.getConfig(editor.getPath());
+        var exclude = globule.isMatch(config && config.excludeFiles, this.getFilePath(editor.getPath()));
+
+        if (grammarScopes.indexOf(editor.getGrammar().scopeName) !== -1 && this.fixOnSave && !exclude) {
+          process.nextTick(() => {
+            this.fixString();
```

Comment:
This re-introduces a race condition, you should pass the editor to the method here instead of getting editor there.
Also, is there any particular reason for running this on nextTick?